### PR TITLE
Add tsconfig and types so editor can detect basic type errors

### PR DESCRIPTION
**Pull request by lukaw3d**

### CODE REVIEW

File: src/popup/pages/Networks/index.js [hunk 1]
```diff
@@ -63,16 +63,9 @@ class NetworkPage extends React.Component {
...
-    netConfig: state.network,
-});
-
+const mapStateToProps = (state) => ({});
 function mapDispatchToProps(dispatch) {
-    return {
-        updateNetConfigList: (config) => {
-            dispatch(updateNetConfigList(config))
-        },
-    };
```
Comment (pro-wh) [inline src/popup/pages/Networks/index.js:75]: what are these fixes? were these fields unused?

**Review by pro-wh — COMMENTED** [review]
looks agreeable mostly. one question about a removal (inline)

could I have you add a CI step to run the type check?

File: src/popup/pages/Networks/index.js (hunk 1 above)
Comment (lukaw3d) [reply src/popup/pages/Networks/index.js:75]: Inner updateNetConfigList is not defined, and luckily whole thing was unused

**Review by pro-wh — APPROVED** [review]

**Comment by lukaw3d** [comment]
CI type check would find hundreds of remaining errors :D

File: src/popup/pages/Networks/index.js (hunk 1 above)
Comment (pro-wh) [reply src/popup/pages/Networks/index.js:75]: ah yeah we just set it once in the constructor

**Comment by pro-wh** [comment]
are they false positives?

**Comment by lukaw3d** [comment]
Some are false positives; many are because types are missing; and a lot are bad but ignorable.

Without CI and without making this part of build process will keep them in editor only for now

**Pull request merged**